12_glob_lab/globlab.txt

```diff
@@ -5,25 +5,34 @@ https://www.tocode.co.il/bundles/unix1/lessons/globlab
 
   1. All files whose name starts with a lowercase letter
   Answer: 
-
+ls [a-z]*
   2. All files whose name starts with an uppercase letter
   Answer:
-
+ls [A-Z]*
   3. All files whose name ends in an "a", "e", "i", "o" or "u"
   Answer:
 
+  leo@leo-VirtualBox:~/lab$ ls *.[a,e,i,o,u]
+
   4. All files whose name DOES NOT end in an "a", "e", "i", "o" or "u"
   Answer:
+leo@leo-VirtualBox:~/lab$ ls *.[!a,e,i,o,u]
 
   5. All files with extension ".c" that are in a folder named "src"
   Answer:
-
+ls *.c src/
   6. All files with extension ".c" under any folder
   Answer:
-
+find . -type f -name *.c
```

Comment:
almost! after reading my comment above let's look at ```*.c``` find will never see ```*.c``` here try this:
```
vitali@Tux:~/work/react-hut$ echo -type f -name *.c
-type f -name *.c

vitali@Tux:~/work/react-hut$ touch crap-I-was-matched-by-a-glob.c

vitali@Tux:~/work/react-hut$ echo -type f -name *.c
-type f -name crap-I-was-matched-by-a-glob.c

vitali@Tux:~/work/react-hut$ echo -type f -name "*.c"
-type f -name *.c

```

yeah I know... very confusing... the golden rule is add quotes if you want the command to see it :)